### ISSUE TITLE
[Bug]: loading big data(220+G) out of memory 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#8040]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):fb62c0e6960af6eedbdab06adecb2ebf7d97e8ba
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

when loading big data ,the mo sometimes crashed by out of memory,
error log:
2023/02/15 13:58:36.492879 +0800 INFO mpool/mpool.go:84 MPool pipeline-c7b7cc0e-acf0-11ed-b5a4-b07b25f8b524 new high watermark
     allocations : 121643199
     frees : 116599855
     go runtime allocations : 0
     alloc bytes : 6079734731373
     free bytes : 4982370239889
     current bytes : 1097364491819
     high water mark : 1097364492076

2023/02/15 13:58:36.504368 +0800 INFO mpool/mpool.go:84 MPool global new high watermark
     allocations : 151777603
     frees : 145523831
     go runtime allocations : 0
     alloc bytes : 6766109761892
     free bytes : 5666598150030
     current bytes : 1099511611862
     high water mark : 1099511633622

2023/02/15 13:58:36.504467 +0800 ERROR mpool/mpool.go:527 error: error: out of memory
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Alloc
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:527
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Realloc
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:610
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Grow
        /data1/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/common/mpool/mpool.go:676
github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).extend

**Comment by nnsgmsone** [comment]
I'll be at 0.8 fix.

**Comment by nnsgmsone** [comment]
I'll be at 0.8 fix.

**Comment by nnsgmsone** [comment]
转成feature #388把 @aressu1985

**Comment by nnsgmsone** [comment]
转成feature #388把 @aressu1985

**Comment by nnsgmsone** [comment]
转成feature #388把 @aressu1985

**Comment by nnsgmsone** [comment]
转成feature #388把 @aressu1985

**Comment by nnsgmsone** [comment]
转成feature #388把 @aressu1985

**Comment by nnsgmsone** [comment]
I will deal with the memory issue first

**Comment by nnsgmsone** [comment]
https://github.com/matrixorigin/matrixone/issues/8042

**Comment by nnsgmsone** [comment]
I will deal with the memory issue first

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
等于https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
fixed by #9069

**Comment by nnsgmsone** [comment]
fixed by #9069

**Comment by aressu1985** [comment]
fixed
